io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -221,9 +221,17 @@ impl Share for Lvol {
     }
 
     async fn update_properties<P: Into<Option<UpdateProps>>>(
-        self: Pin<&mut Self>,
+        mut self: Pin<&mut Self>,
         props: P,
     ) -> Result<(), Self::Error> {
+        let props = UpdateProps::from(props.into());
+        let allowed_hosts = props.allowed_hosts().clone();
+        // Set but don't sync in lvol metadata. Sync will happen in case the
+        // node reboots e.g during upgrade.
+        self.as_mut()
+            .set_no_sync(PropValue::AllowedHosts(allowed_hosts))
+            .await?;
```

Comment:
Can we check what the current stored value is? If it's the same then we don't need to call set?

Reply:
> Can we check what the current stored value is? If it's the same then we don't need to call set?

get() call will read from the disk so will be expensive. If we **have** to go to disk we better do set(), rather than get() + conditional set_no_sync(). Looks unnecessary to me.

Reply:
IIRC the metadata is kept in memory

Reply:
> IIRC the metadata is kept in memory

Yes a copy is kept in memory. I took this comment at face value :) 
```
/// Get/Read a property from this lvol from disk.
async fn get(&self, prop: PropName) -> Result<PropValue, Error> {
```
But yes, spdk API is giving the in-core value and not fetching from disk. Let me see, I can put a `get` check.

Reply:
Please fix the comment while you're at it :)

Reply:
Though I'm still a bit wary about putting a `get` before `set_no_sync`. We don't do that anywhere today, so let's be consistent?

Reply:
Why don't we change the set_no_sync to do this internally? Check if current value is same, then don't do anything

Reply:
> Why don't we change the set_no_sync to do this internally? Check if current value is same, then don't do anything

That can be done. For properties `Shared` and `EntityId` it'll be simpler, for `AllowedHosts` a bit verbose because it's a Vec<String>. Looks ok?

```
     fn set_no_sync()
           [...]
            PropValue::AllowedHosts(mut hosts) => {
                let name = PropName::from(&prop).to_string().into_cstring();
                let value = hosts.join(",").into_cstring();
                if let Ok(pval) = self.get(PropName::AllowedHosts).await {
                    match pval {
                        PropValue::AllowedHosts(mut hlist) => {
                            if hlist.len() == hosts.len() {
                                hosts.sort(); hlist.sort();
                                if hosts == hlist {
                                    return Ok(())
                                }
                            }
                        },
                        _ => (),
                    }
                };
```

Reply:
Seems like it yeah

Reply:
@tiagolobocastro PTAL at latest diff.